lib/smart_proxy_ansible/runner/ansible_runner.rb

```diff
@@ -34,10 +34,23 @@ def start
         start_ansible_runner
       end
 
+      def initialize_command(*command)
+        r, w = IO.pipe
+
+        @command_pid = spawn(*command, :out => w, :err => w, :in => '/dev/null')
+        @command_out = r
+        w.close
+      rescue Errno::ENOENT => e
+        publish_exception("Error running command '#{command.join(' ')}'", e)
+      end
+
       def refresh
-        return unless super
+        super
```

Comment:
Does super do something useful for us? Is it to close the Pipe when the command finishes?

Reply:
It watches the process running as `@command_pid`, reads its output (which we later discard) and finally reaps the process